Group pages and modals into packages

oshorefueled opened the pull request:
This PRs groups pages and modals into their own packages. It also adds a load package which would replace the common struct shared among pages, modals and a number of other godcr components.

oshorefueled committed:
move create password modal to modal package

oshorefueled committed:
move modals to modal package

oshorefueled committed:
add load package

* create modal and page packages and move modals and pages into
  them respectively.
* add a load package. Load replaces the common type and holds data
  that is shared among ui components in the ui package.

oshorefueled committed:
cleanup errors on main page

oshorefueled committed:
move pages from ui to page package

- add comments to the load package and utils file
- move help page to page package
- move sign message page to page package
- move receive page to page package
- move security tools, validate and verify message pages
- move settings page to page package
- move the more page to page package
- move overview and transaction pages to page package

oshorefueled committed:
fix wrong call on theme in statistics page

- add transactions page to case for sending wallet notifications
- go fmt load.go

oshorefueled committed:
resolve crash when internet connection is off

oshorefueled committed:
resolve go lint issues

beansgum commented on ui/create_restore_page.go:
```diff
@@ -645,7 +645,7 @@ func (pg *createRestore) handle() {
 			} else {
 				pg.common.wallet.SetupListeners()
-				pg.common.changeWindowPage(newMainPage(pg.common), false)
+				pg.common.changeWindowPage(newMainPage(pg.common, nil), false)
```
`nil` value passed for the load parameter can cause a nil pointer exception.

beansgum commented on ui/main_page.go:
```diff
@@ -221,7 +225,11 @@ func (mp *mainPage) unlockWalletForSyncing(wal *dcrlibwallet.Wallet) {
+
+	// TODO: This function should be only called when
+	// dcrlibwallet update notifications are receieved
+	mp.updateBalance()
```
This probably got in from rebase, the `updateBalance` function have been moved to on resume and is called when notifications are received.

beansgum commented on ui/main_page.go:
```diff
@@ -34,31 +37,36 @@ type mainPage struct {
-	common.returnPage = &mp.previous
-	common.page = &mp.current
-	common.toggleSync = func() {
+	// todo: common methods will be removed when
```
Typo in comment.

beansgum commented on ui/main_page.go:
```diff
@@ -268,9 +274,14 @@ func (mp *mainPage) changeFragment(page Page, id string) {
 	mp.changePage(id)
 }
 
+func (mp *mainPage) _changeFragment(page interface{}, id string) {
```
I think it's better to move the `Page` interface to the page package and let the pages in this package import it to avoid this anonymous cast.

beansgum commented on ui/markdown.go:
```diff
@@ -0,0 +1,546 @@
+package ui
```
Delete this file, it has been moved to the `ui/renderers` package.

beansgum commented on ui/modal/info_modal.go:
```diff
@@ -0,0 +1,224 @@
+
+const Info = "info_modal"
+
+type Common interface {
```
Remove unused interface.

beansgum commented on ui/modal/modal.go:
```diff
@@ -0,0 +1,56 @@
+package modal
+
```
This file does not contain any model-specific content, I guess it can be renamed to `utils.go`.

beansgum commented on ui/modal_templates.go:
```diff
@@ -23,26 +22,23 @@ const (
-					</span>`
-
-			return renderers.RenderHTML(text, th).Layout(gtx)
+			text := th.Body1("After you or your counterparty has genrated a signature, you can use this form to verify the" +
```
Fix error from rebasing, this text and the text in other functions in this file have been replaced with HTML.

beansgum commented on ui/main_page.go:
```diff
@@ -133,11 +141,7 @@ func (mp *mainPage) OnResume() {
-	} else {
-		mp.changeFragment(OverviewPage(mp.pageCommon), PageOverview)
-	}
+	mp.changeFragment(page.NewOverviewPage(mp.load), page.Overview)
```
Check for the current page before switching to overview.
```suggestion
	if pg, ok := mp.pages[mp.current]; ok {
		pg.OnResume()
	} else {
		mp.changeFragment(OverviewPage(mp.pageCommon), PageOverview)
	}
```

beansgum commented on ui/page.go:
```diff
@@ -260,29 +262,30 @@ func (win *Window) newPageCommon(decredIcons map[string]image.Image) *pageCommon
 
-	pages[PageMore] = MorePage(common)
-	pages[PageReceive] = ReceivePage(common)
+	pages[PageWallet] = WalletPage(common, l)
```
Wallets page isn't part of this list.

beansgum commented on ui/main_page.go:
```diff
@@ -241,21 +249,19 @@ func (mp *mainPage) handle() {
-				mp.changeFragment(TransactionsPage(mp.pageCommon), PageTransactions)
-			} else if i == 2 {
-				mp.changeFragment(WalletPage(mp.pageCommon), PageWallet)
+				mp.changeFragment(page.NewTransactionsPage(mp.load), page.Transactions)
```
Wallets page was included here and has been removed.

beansgum commented on ui/page/about_page.go:
```diff
@@ -1,18 +1,18 @@
 )
 
-const PageAbout = "About"
+const About = "About"
```
Why was the `Page` prefix remove from the page's name constant? This would get confusing as this constant will always show up when typing `About-`. I'd recommend that prefix should be restored or it can be changed to something better like `AboutPageID` since the constants are used as ids.

beansgum commented on ui/page/components.go:
```diff
@@ -0,0 +1,619 @@
+	purchasingAccountTitle = "Purchasing account"
+	sendingAccountTitle    = "Sending account"
+	receivingAccountTitle  = "Receiving account"
+)
```
Remove unused constants.

beansgum commented on ui/page/components.go:
```diff
@@ -0,0 +1,619 @@
+
+// layoutBalance aligns the main and sub DCR balances horizontally, putting the sub
+// balance at the baseline of the row.
+//func (page *pageCommon) layoutBalance(gtx layout.Context, amount string, isSwitchColor bool) layout.Dimensions {
```
Delete unused codes.

beansgum commented on ui/page/components.go:
```diff
@@ -0,0 +1,619 @@
+	)
+}*/
+
+func displayToast(th *decredmaterial.Theme, gtx layout.Context, t *load.Toast) layout.Dimensions {
```
Remove unused function, this file needs cleanup.

oshorefueled replied:
some methods were left commented or unused because they'll be called the pages calling them have been migrated to the page package.

beansgum commented on ui/page/markdown.go:
```diff
@@ -0,0 +1,546 @@
+package page
+
```
Not sure how this file got there but it doesn't belong to this folder.

beansgum commented on ui/page/more_page.go:
```diff
@@ -114,10 +114,11 @@ func (pg *morePage) layoutMoreItems(gtx layout.Context, common *pageCommon) layo
-															page = "Security Tools"
-														}
-														return common.theme.Body1(page).Layout(gtx)
+														// todo: uncomment when security tools has been moved to the page package
```
Security tools have been moved to the page package, time to uncomment.

beansgum commented on ui/page/page.go:
```diff
@@ -0,0 +1,228 @@
+}
+
+type WalletAccountSelector struct {
+	title                     string
```
Remove unused structs.

beansgum commented on ui/page/transactions_page.go:
```diff
@@ -72,14 +73,15 @@ func TransactionsPage(common *pageCommon) Page {
-	pg.wallets = pg.sortedWalletList()
-	pg.createOrUpdateWalletDropDown(&pg.walletDropDown, pg.wallets)
+func (pg *TransactionsPage) OnResume() {
+	log.Infof("WALLETS GUY! MULTI %v\n", pg.Load)
```
😁

beansgum commented on ui/page/verify_message_page.go:
```diff
@@ -147,55 +147,53 @@ func (pg *verifyMessagePage) verifyMessageResponse() layout.Widget {
 			pg.verifyMessage.Text = ""
 			pg.verifyMessageStatus = nil
-			valid, err := c.wallet.VerifyMessage(pg.addressInput.Editor.Text(), pg.messageInput.Editor.Text(), pg.signInput.Editor.Text())
+			valid, err := pg.WL.Wallet.VerifyMessage(pg.addressInput.Editor.Text(), pg.messageInput.Editor.Text(), pg.signInput.Editor.Text())
```
```suggestion
			valid, err := pg.WL.MultiWallet.VerifyMessage(pg.addressInput.Editor.Text(), pg.messageInput.Editor.Text(), pg.signInput.Editor.Text())
```

beansgum commented on ui/send_page.go:
```diff
@@ -2,6 +2,7 @@ package ui
 
 import (
 	"fmt"
+
```
Cleanup imports.

beansgum commented on ui/window.go:
```diff
@@ -110,20 +112,91 @@ func CreateWindow(wal *wallet.Wallet, decredIcons map[string]image.Image, collec
+	//	selectedUTXO: make(map[int]map[int32]map[string]*wallet.UnspentOutput),
+	//	toast:        &win.toast,
+	//	internalLog:  &win.internalLog,
+	//}
```
Delete.

oshorefueled committed:
move account details, restore wallet, wallet page to page package

- move seed backup page to the page package

oshorefueled committed:
add interface file to the load package

- replace parameters in window methods from interface type to Modal and Page types

oshorefueled committed:
delete unused modals in the ui package

beansgum commented on ui/page/verify_message_page.go:
```diff
@@ -46,35 +46,35 @@ func VerifyMessagePage(c *pageCommon) Page {
+				pg.clearInputs()
+				// todo: uncomment when the Wallet page has been moved
+				// pg.ChangePage(Wallet)
+				pg.ChangePage(*pg.ReturnPage)
```
Wallet page have been moved.

beansgum commented on ui/page/verify_message_page.go:
```diff
@@ -46,35 +46,35 @@ func VerifyMessagePage(c *pageCommon) Page {
-	if *c.returnPage == PageSecurityTools {
+func (pg *VerifyMessagePage) Layout(gtx layout.Context) layout.Dimensions {
+	var walletName = pg.WL.Info.Wallets[*pg.SelectedWallet].Name
+	if *pg.ReturnPage == SecurityTools {
```
Remove this, verify message page isn't available in wallet's option.

beansgum commented on ui/page/page.go:
```diff
@@ -0,0 +1,238 @@
+	receivingAccountBtn       *widget.Clickable
+	purchaseTicketAccountBtn  *widget.Clickable
+	sendOption                string
+	walletInfoButton          decredmaterial.IconButton
```
Delete unused structs.

oshorefueled committed:
move license page to the page package

oshorefueled committed:
uncomment code related to pages that have been moved

- append PageID as a Suffix to page constants
- restore html code which were removed from modal template file during rebase
- remove unused structs

beansgum commented on ui/page/license_page.go:
```diff
@@ -1,14 +1,16 @@
+const License = "License"
 
-const License = `ISC License
+const license = `ISC LicenseText
```
```suggestion
const license = `ISC License
```

beansgum commented on ui/page/wallet_page.go:
```diff
@@ -16,7 +19,7 @@ import (
 )
 
-const PageWallet = "Wallets"
+const Wallet = "Wallets"
```
```suggestion
const WalletPageID = "Wallets"
```

beansgum commented on ui/privacy_page.go:
```diff
@@ -3,10 +3,13 @@ package ui
 import (
 	"fmt"
 
+	"github.com/planetdecred/godcr/ui/page"
```
Cleanup imports

beansgum commented on ui/window.go:
```diff
@@ -6,6 +6,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/planetdecred/godcr/ui/load"
```
Cleanup imports

oshorefueled committed:
re-arrange package imports

- rebase on master branch
- change wallet page ID constant to WalletPageID
